tutorials/misc/plot_ecog.py

```diff
@@ -169,34 +151,133 @@ def animate(i, activity):
+
+# Apply a high pass filter to get rid of drift, decrease time range
+raw_ecog.filter(l_freq=5, h_freq=None)
+
+# Downsample again, to compute the animation faster
+sfreq = 50  # Hz
+raw_ecog.resample(sfreq)
+ts_data = raw_ecog.get_data()
+
+# recompute events for new sampling frequency
+events, event_id = mne.events_from_annotations(raw_ecog)
+onset_events = events[events[:, 2] == event_id['onset']]
+
+# invert the event_ids so that we can look up by id and get the name
+# to display on the plot
+inv_event_id = {v: k for k, v in event_id.items()}
+
+# Use one second after the seizure onset as the animation start,
+# right as the seizure is ramping up
+start_sample = int(onset_events[0, 0] + sfreq)
+
+
+# Create an initialization and animation function to pass to FuncAnimation.
+# This time we will also pass the plot title so that can be updated
+# with information from the annotations.
+def init():
+    """Create an empty frame."""
+    return paths, title, *tpaths
+
+
+def animate(i, activity, events):
+    """Animate the plot."""
+    paths.set_array(activity[:, i])
+    tsl = slice(i + start_sample - sfreq, i + start_sample + sfreq)
+    for j, tpath in enumerate(tpaths):
+        tpath.set_data(range(2 * sfreq), ts_data[j, tsl])
+    # If this sample contains an annotation (for a seizure-related behavior)
+    # then change the title of the plot
+    if i + start_sample in events[:, 0]:
+        # Currently this doesn't replace the text, but writes over it.
+        # This needs fixing
+        event_idx = np.argwhere(events[:, 0] == (i + start_sample))[0][0]
+        title.set_text(inv_event_id[events[event_idx, 2]])
+        fig.canvas.draw()  # force redrawing
+    return paths, title, *tpaths
+
+
+# create the figure and apply the animation of the
+# raw time series data
+fig, ax = plt.subplots(figsize=(5, 5))
+tax = fig.add_axes([0.125, 0.02, 0.6, 0.15])
+ax.imshow(im)
+ax.set_axis_off()
+tax.set_axis_off()
+
+# We want the mid-point (0 uV) to be white, so we will scale from -vmax to vmax
+# so that negative voltages are blue and positive voltages are red
+ts_data = raw_ecog.get_data()
+vmax = np.percentile(ts_data, 90)
+paths = ax.scatter(*xy_pts.T, c=np.zeros(len(xy_pts)), s=100,
+                   cmap=cmap, vmin=-vmax, vmax=vmax)
+ax.set_xlim([0, im.shape[0]])
+ax.set_ylim([im.shape[1], 0])
+tsl = slice(start_sample - sfreq, start_sample + sfreq)
+tpaths = tax.plot(np.zeros((ts_data.shape[0], 2 * sfreq)).T)
+tax.plot([sfreq, sfreq], [ts_data.min(), ts_data.max()], color='k')
+fig.colorbar(paths, ax=ax)
+title = ax.set_title('iEEG voltage over time', size='large')
+
+# this will be a much longer animation because the seizure is quite long
+sl = slice(start_sample, ts_data.shape[1])
+show_power = ts_data[:, sl]
+anim = animation.FuncAnimation(fig, animate, init_func=init,
+                               fargs=(show_power, events),
+                               frames=show_power.shape[1] - sfreq,
```

Comment:
This is the problem, it converts `frames` from an int to a float and for some reason the behaviors are different. Maybe it means "seconds" when you use it that way? I don't see it in their docs, but changing it to an int seemed to fix it